db.go

```diff
@@ -1370,3 +1387,63 @@ func (db *DB) DropPrefix(prefix []byte) error {
 	db.opt.Infof("DropPrefix done")
 	return nil
 }
+
+// Subscribe can be used watch key changes for the given key prefix
+func (db *DB) Subscribe(prefix []byte, cb callback) (func(), error) {
+	if cb == nil {
+		return nil, errors.New("callback can't be nil")
```

Comment:
It is generally a good idea to declare the error once and return the same pointer every time.